src/dashboard/scripts/packs.ts

```diff
@@ -0,0 +1,186 @@
+}
+
+function getPackIcon(packid: string) {
+    const pack = packData.packs.find((pack) => pack.id === packid);
+    if (!pack) {
+        throw new Error("Pack not found");
+    }
+
+    return `${cdn}/icons/${pack.icon}`;
+}
+
+// render all packs
+for (const pack of packData.packs) {
+    const icon = getPackIcon(pack.id);
+    const packElement = document.createElement("div");
+    const description = marked.parse(pack.description, { async: false }) as string;
+    packElement.innerHTML = `
+        <img src="${icon}" alt="${pack.friendly_name}">
+        <h2>${pack.friendly_name}</h2>
+        <div>${description}</div>
+    `;
+
+    // dinamically load the download buttons
+    for (const version of <const>["1.8.9", "1.20.5", "bedrock"]) {
+        if (!pack.downloads[version]) {
+            continue;
+        }
+
+        const downloadButton = document.createElement("button");
+        downloadButton.textContent = `Download for ${version === "bedrock" ? "Bedrock" : version}`;
+        downloadButton.addEventListener("click", () => downloadPack(pack.id, version));
+        packElement.appendChild(downloadButton);
+    }
+    document.body.appendChild(packElement);
+}
+
+const commentForm = document.getElementById("commentForm") as HTMLFormElement;
+// add "log in to comment" warning
+app.api.user.get().then((response) => {
+    if (response.status !== 200) {
+        // remove the submit button
+        commentForm.querySelector("button")?.remove();
+
+        const warning = document.createElement("a");
+        warning.innerText = "You must be logged in to comment!";
+        warning.href = "/api/auth?redirect=/packs.html";
+        commentForm.append(warning);
+    }
+});
+
+// add select menu for all packs
+const select = document.createElement("select");
+select.name = "packid";
+for (const pack of packData.packs) {
+    const option = document.createElement("option");
+    option.value = pack.id;
+    option.innerText = pack.friendly_name;
+    select.appendChild(option);
+}
+commentForm.prepend(select);
+
+export function sendComment() {
+    // create a form data object
+    const formData = new FormData(commentForm);
+
+    // send the comment to the server
+    app.api.comments
+        .post({
+            packid: formData.get("packid") as string,
+            comment: formData.get("comment") as string,
+            "h-captcha-response": formData.get("h-captcha-response") as string
+        })
+        .then((res) => {
+            if (res.status === 200) {
+                // TODO: show some kind of modal ("your comment has been sent to review" or smth)
+                return;
+            }
+            if (res.status === 422) {
+                // TODO: badly formatted comment (maybe user was a nerd and tried to bypass textarea length limit)
+                return;
+            }
+            if (res.status === 401) {
+                // TODO: invalid captcha
+                return;
+            }
```

Comment:
@noClaps here are some modal suggestions for posting a comment, so the user gets some sort of feedback
```suggestion
    if (res.status === 200) {
        // TODO: show some kind of modal ("your comment has been sent to review" or smth)
        return;
    }
    if (res.status === 422) {
        // TODO: badly formatted comment (maybe user was a nerd and tried to bypass textarea length limit)
        return;
    }
    if (res.status === 401) {
        // TODO: invalid captcha
        return;
    }
```